google/cloud/aiplatform/jobs.py

```diff
@@ -1403,6 +1406,29 @@ def from_local_script(
                         {"name": key, "value": value}
                         for key, value in environment_variables.items()
                     ]
+            else:
```

Comment:
Docstrings need to be updated explaining this behavior and the customer container requirements.